metadataserver/cni/env.go

```diff
@@ -0,0 +1,249 @@
+// +build linux
+
+package cni
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/Netflix/titus-executor/api/netflix/titus"
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/containernetworking/cni/pkg/version"
+
+	"github.com/Netflix/titus-executor/utils"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	tt "github.com/Netflix/titus-executor/executor/runtime/types"
+	mt "github.com/Netflix/titus-executor/metadataserver/types"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	v1 "k8s.io/api/core/v1"
+)
+
+const kubeletAPIPodsURL = "https://localhost:10250/pods"
+
+func getPrev(args *skel.CmdArgs) (*current.Result, error) {
+	var cfg types.NetConf
+
+	err := json.Unmarshal(args.StdinData, &cfg)
+	if err != nil {
+		err = errors.Wrap(err, "Cannot parse cni configuration")
+		return nil, err
+	}
+
+	err = version.ParsePrevResult(&cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "error parsing prevResult")
+	}
+
+	prev, err := current.NewResultFromResult(cfg.PrevResult)
+	if err != nil {
+		return nil, errors.Wrap(err, "error casting prevResult")
+	}
+	logrus.Debugf("prevResult %#v", prev)
+
+	return prev, err
+}
+
+func getPodName(args *skel.CmdArgs) (string, error) {
+	var k8sArgs utils.K8sArgs
+	err := types.LoadArgs(args.Args, &k8sArgs)
+	if err != nil {
+		return "pod-name", errors.Wrap(err, "Unable to parse CNI k8s args")
+	}
+	logrus.Debugf("k8s args %#v", k8sArgs)
+
+	return string(k8sArgs.K8S_POD_NAME), nil
+}
+
+func getPod(args *skel.CmdArgs) (*v1.Pod, error) {
+	var k8sArgs utils.K8sArgs
+	err := types.LoadArgs(args.Args, &k8sArgs)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to parse CNI k8s args")
+	}
+	logrus.Debugf("k8s args %#v", k8sArgs)
+
+	ctx := context.Background()
+	defer ctx.Done()
```

Comment:
Why `defer ctx.Done()`?
Should this be:

```go
ctx, cancel := context.WithCancel(context.Background())
defer cancel()
```

Reply:
Yup, thanks.